File: Vorbereitung/Übungen/2/2-5.cs
namespace Uebung;

class Program
{
    static void Main(string [] args)
    {
        int x = Console.ReadLine();
        Console.WriteLine($"Die Fakultät von {x} ist {Fac(x)}");
    }
    static int Fac(int x)
    {
        if (x == 1 || x == 0)
        {
            return 0;
        }
        if (x < 0)
        {
            Console.WriteLine("Fakultät für negative Werte nicht definiert.");
            Environment.Exit(0);
        }
        return x * Fac(x-1);
    }
}

File: Vorbereitung/Übungen/2/2-5.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int num;
    scanf("%d", &num);
    print("Die Fakultät von %d ist %d", num, fac(num));
    return 0;
}
int fac(x) {
    if (x == 1 || x == 0) {
        return 0;
    }
    if (x < 0) {
        print("Für negative Werte nicht definiert");
        exit(EXIT_SUCCESS);
    }
    return x * fac(x-1);
}